### PR TITLE
Soft Margin Loss function

### PR DESCRIPTION
**Pull request by iamshnoo**
Relevant issue : #2200 
PyTorch docs : [here](https://pytorch.org/docs/stable/nn.html#softmarginloss)

To compare bare-bones armadillo implementation with PyTorch implementation, see this [Google Colab notebook](https://colab.research.google.com/drive/1W4ndom2jMmsGhraNf17VacOzk2Q748k8?usp=sharing).

### CODE REVIEW

**Comment by zoq** [comment]
Looks like the notebook is private?

**Comment by iamshnoo** [comment]
> Looks like the notebook is private?

Updated the link. Thanks!

**Comment by iamshnoo** [comment]
Thanks for reviewing and merging this!
